scandium-core/api-changes.json

```diff
@@ -144,6 +162,12 @@
 					"old": "method void org.eclipse.californium.scandium.dtls.SupportedPointFormatsExtension::addECPointFormat(org.eclipse.californium.scandium.dtls.SupportedPointFormatsExtension.ECPointFormat)",
 					"justification": "not part of the public API - removed requires explicit ignore",
 					"oldArchive": "org.eclipse.californium:scandium:jar:2.1.0"
+				},
+				{
+					"code": "java.class.removed",
+					"old": "class org.eclipse.californium.scandium.dtls.PskUtil",
+					"justification": "not part of the public API - obsolete",
+					"oldArchive": "org.eclipse.californium:scandium:jar:2.1.0"
```

Comment:
I'm not sure, what the benefit should be. For me that would look like: you don't get errors on compile time, you don't get error reports on just running old code with the new advanced psk implementation. But it doesn't work. For me, these reported errors are really helpful to save time. The user will get aware, that they use some internal stuff and it's better to adapt their stuff to the new implementation.
That may raise the question again, if a async psk store should be a major version. For me, as I want that feature soon, I assume, more benefit from that advanced psk store in a 2.3 (instead of waiting for 3.0) than having trouble with that change.

Reply:
Imagine a user use this class for something (I don't know why?)
We let the class as he can still use it on its own but we deprecated it to show to user this is no more used in scandium and will probably be deleted in the next major release.

I have exactly the same situation with leshan recently (for the 1.1).
I replaced the static RedisLock by a dynamic JedisLock. Redislock class is no more used in Leshan code but I keep it just in case.
And in this case, I know that at sierra some people just use the RedisLock class but I could have been not aware of this.

> That may raise the question again, if a async psk store should be a major version. For me, as I want that feature soon, I assume, more benefit from that advanced psk store in a 2.3 (instead of waiting for 3.0) than having trouble with that change.

This was not the question I wanted to raise.
Even if I don't agree totally, I understand your position : "estimating what we could break or not break based on your intuition about what is used by user."

My point here "this easy to not break, so why not ?"

Reply:
> My point here "this easy to not break, so why not ?"

I consider using that PskUtil as using scandium by patching. 
So my concern here is, that it may compile, but will not work.

Reply:
It's here we disagree. I consider that I can not know what user do with my code and if I can avoid a break easily I do it. (even if in practice, I also think there poor chance that this code is reused but I also know that sometime my assumption are wrong)

Anyway, I didn't want to start another endless discussion about that.
So let it like this. I asked just in case without so much hope ^^